### PR TITLE
[dom-gpu] Add support for XALT installation in production.sh

### PR DESCRIPTION
**Pull request by victorusu**
One cannot have xalt loaded before installing xalt...

### CODE REVIEW

File: easybuild/easyconfigs/x/xalt/xalt-2.7.24.eb [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+local_transmission = "curl"
+
+# generic configurations
+configopts  = "--with-trackScalarPrgms=yes "
+configopts += "--with-computeSHA1=no "
+configopts += "--with-MySQL=no "
+configopts += "--with-tmpdir=${HOME}/.xalt.d/tmpdir "
+# system specific configurations
+configopts += "--with-trackGPU=yes "
+configopts += "--with-systemPath=/apps/daint/system/bin:/apps/dom/system/bin:/apps/common/system/bin::/usr/local/bin:/usr/bin:/bin:/usr/bin/X11 "
```
Comment (jgphpc) [inline easybuild/easyconfigs/x/xalt/xalt-2.7.24.eb]: can we avoid to depend on /system ?

File: easybuild/easyconfigs/x/xalt/xalt-2.7.24.eb [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+preinstallopts = 'LDFLAGS="-L${EBROOTCUDA}/lib64" CPPFLAGS="-I${EBROOTCUDA}/include" '
+
+modextravars = {
+    'XALT_EXECUTABLE_TRACKING': 'yes',
+    'XALT_SCALAR_SAMPLING': 'no',
+    'XALT_GPU_TRACKING': 'no',
+    'XALT_DIR': '%(installdir)s',
+    'XALT_ETC_DIR': '%(installdir)s/etc',
+    'XALT_TRANSMISSION_STYLE': '%s' % local_transmission,
+    #'XALT_LOGGING_URL': 'asdfasdfasdf',
```
Comment (jgphpc) [inline easybuild/easyconfigs/x/xalt/xalt-2.7.24.eb]: ?

File: easybuild/easyconfigs/x/xalt/xalt-2.7.24.eb [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+configopts += "--with-tmpdir=${HOME}/.xalt.d/tmpdir "
+# system specific configurations
+configopts += "--with-trackGPU=yes "
+configopts += "--with-systemPath=/apps/daint/system/bin:/apps/dom/system/bin:/apps/common/system/bin::/usr/local/bin:/usr/bin:/bin:/usr/bin/X11 "
+configopts += "--with-config=Config/Daint_config.py "
+configopts += "--with-syshostConfig=read_file:/etc/xthostname "
+configopts += "--with-transmission=%s " % local_transmission
+# this is workaround to do not make public the logging_url data
+configopts += "--with-xaltLoggingUrl=`cat /apps/common/easybuild/sources/x/xalt/logging_url.txt` "
+# this is a limitation on EB
```
Comment (jgphpc) [inline easybuild/easyconfigs/x/xalt/xalt-2.7.24.eb]: can you explain more ? why is it needed if gpu_tracking is off ?

File: easybuild/easyconfigs/x/xalt/xalt-2.7.24.eb [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+modextrapaths = {
+    'LD_PRELOAD': 'lib64/libxalt_init.so',
+    'COMPILER_PATH': 'bin'
+}
+
+builddependencies = [
+    ('cudatoolkit/10.1.105_3.27-7.0.1.1_4.1__ga311ce7', EXTERNAL_MODULE),
+]
+dependencies = [
```
Comment (jgphpc) [inline easybuild/easyconfigs/x/xalt/xalt-2.7.24.eb]: dependencies list is also a builddep no (hence builddep is not needed) ?

File: easybuild/easyconfigs/x/xalt/xalt-2.7.24.eb (hunk 1 above)
Comment (victorusu) [reply easybuild/easyconfigs/x/xalt/xalt-2.7.24.eb]: Unfortunately, no!

File: easybuild/easyconfigs/x/xalt/xalt-2.7.24.eb (hunk 2 above)
Comment (victorusu) [reply easybuild/easyconfigs/x/xalt/xalt-2.7.24.eb]: We do not need to define here, but I kept because it is part of our investigation.

**Pull request merged**